src/threading/SerialTypes.h

```diff
@@ -126,6 +126,7 @@ struct Value {
 		vec_t vector_val;
 		addr_t addr_val;
 		subnet_t subnet_val;
+		const char* pattern_text_val;
```

Comment:
May make sense to just use the existing `string_val` union member since we already use it for storing several types: looks like enums, strings, files, and functions.  Of course, that would mean adapting the nul-terminated assumption in the other parts of the patch.

Reply:
I thought about this as well. Thought you might point it out :) 

What made me go with this way was that when you're building a new pattern's `RE` matcher it expects a null terminated c-string like this while `string_val`s current usage seems to assume that it is not null terminated. Rather than adding a null-terminator when building the pattern or sometimes having `string_val`s that are null terminated I thought it would be cleaner to just have a separate value in the struct.